datafusion/physical-expr/src/expressions/binary.rs

```diff
@@ -1255,14 +1235,25 @@ impl BinaryExpr {
         scalar: &ScalarValue,
         array: &ArrayRef,
     ) -> Result<Option<Result<ArrayRef>>> {
+        let bool_type = &DataType::Boolean;
         let scalar_result = match &self.op {
-            Operator::Lt => binary_array_op_scalar!(array, scalar.clone(), gt),
-            Operator::LtEq => binary_array_op_scalar!(array, scalar.clone(), gt_eq),
-            Operator::Gt => binary_array_op_scalar!(array, scalar.clone(), lt),
-            Operator::GtEq => binary_array_op_scalar!(array, scalar.clone(), lt_eq),
-            Operator::Eq => binary_array_op_scalar!(array, scalar.clone(), eq),
+            Operator::Lt => {
+                binary_array_op_dyn_scalar!(array, scalar.clone(), gt, bool_type)
```

Comment:
This is the actual fix -- to use `binary_array_op_dyn_scalar` which @WinkerDu  updated to handle null constants rather than `binary_array_op_scalar` which does not.

I plan to look into removing `binary_array_op_scalar` completely as a follow on PR.

Reply:
Sorry for late response, love this fix ❤️